web-extension-1/src/popup/popup.tsx

```diff
@@ -1 +1,31 @@
-console.log('Popup script');
\ No newline at end of file
+/**
+ * LOADED VIA "manifest.browser_action.default_popup"
+ * BUT USING an HTML file: popup.html
+ *
+ * - Can access all WebExtensions APIs it has permissions for
+ * - Loaded every time the popup is shown
+ * - Unloaded every time the popup is closed
+ */
+
+import { StationMessage } from '@src/shared';
+
+console.log('Popup script');
+
+// 1. Get Active tab Id
+browser.tabs
+  .query({ active: true, currentWindow: true })
+  .then(tabs => {
+    const activeTabId = tabs[0].id;
+    if (activeTabId) {
+      // 2. Send message to background script
+      // to acknowledge popup opening on current tab
+      browser.tabs.sendMessage(activeTabId, {
+        command: 'INCREMENT_TAB_POPUP_OPEN_COUNT_REQUEST',
+        tabId: activeTabId,
+        data: null,
+      } as StationMessage);
+    }
+  })
+  .catch(reason => {
+    console.log('[Active Tab Query] Error: ', reason);
+  });
```

Comment:
popup can (and should) communicate directly with the background page when it can.
Here is how the communication for an increment happens:
```typescript
// Current implementation
popup -> content_script -> background page -> content_script
// Simpler version
popup -> background page -> content_script
```

You can retrieve the active Tab like you did in the popup but directly in the background page.